### PR TITLE
Extend styles with color mixin

### PR DESCRIPTION
**Pull request by Nodarii**

### CODE REVIEW

**Comment by Nodarii** [comment]
I signed it!

**Comment by samccone** [comment]
LGTM thx a ton @nadarik 

CC @notwaldorf

File: iron-icon.html [hunk 1]
```diff
@@ -89,6 +100,7 @@
 
       width: var(--iron-icon-width, 24px);
       height: var(--iron-icon-height, 24px);
+      color: var(--iron-icon-color, inherit);
```
Comment (notwaldorf) [inline iron-icon.html:103]: I thought the color was being given by the `fill: currentcolor;` style right above. Is that still needed? Do they work together?

Comment (Nodarii) [inline iron-icon.html:103]: @notwaldorf, it's not the same.
It works with `fill: currentcolor`. 
Right now we can colorize icons only via some color inheritance or by applying some global class like `.green { color: green }`. But with this change we will be able to apply **isolated**(encapsulated) **custom-styles** just like we do it with size (on the L28)

Comment (notwaldorf) [inline iron-icon.html:103]: I've merged https://github.com/PolymerElements/iron-icon/pull/39 which adds custom properties for `fillColor` and `strokeColor`. Is this still needed?

Comment (Nodarii) [inline iron-icon.html:103]: Cool, it's not needed anymore! Thx

**Comment by notwaldorf** [comment]
Closed as already fixed in a different PR

**Pull request closed without merging**